lib/src/set/set_builder.dart

```diff
@@ -52,7 +52,7 @@ class SetBuilder<E> {
       // Can't use addAll because it requires an Iterable<E>.
       final Set<E> set = new Set<E>();
       for (final element in iterable) {
-        set.add(element);
+        set.add(element as E);
```

Comment:
This should actually be an explicit check I think.

if (element is E) {
  ...
} else {
  throw ...
}